src/HID-APIs/SurfaceDialAPI.hpp

```diff
@@ -37,25 +37,39 @@ void SurfaceDialAPI::begin(void)
 void SurfaceDialAPI::end(void)
 {
 	_button = false;
-	rotate(0);
+	update();
 }
 
 void SurfaceDialAPI::click(void)
 {
 	_button = true;
-	rotate(0);
+	update();
 	_button = false;
-	rotate(0);
+	update();
 }
 
 void SurfaceDialAPI::rotate(int16_t rotation)
+{
+	posrot(rotation, _xAxis, _yAxis);
+}
+
+void SurfaceDialAPI::position(int8_t x, int8_t y)
+{
+	posrot(0, x, y);
+}
+
+void SurfaceDialAPI::posrot(int16_t rotation, int8_t x, int8_t y)
```

Comment:
Please change this name into something more readable if you intend to keep this function